test/test_type_promotion.py

```diff
@@ -167,6 +167,24 @@ def test_half(self, device):
         default_tensor = torch.tensor(100000.0, device=device)
         self.assertEqual((half + default_tensor).dtype, torch.get_default_dtype())
 
+    @float_double_default_dtype
+    def test_bfloat16(self, device):
+        bf = torch.tensor(5.5, dtype=torch.bfloat16, device=device)
+        self.assertEqual((bf + 2.2).dtype, torch.bfloat16)
+        self.assertEqual((bf + 100000).dtype, torch.bfloat16)  # inf
+        default_tensor = torch.tensor(100000.0, device=device)
```

Comment:
You don't really need this test with the default tensor (it will just be float32 or float64, which you're already testing).

This test looks good. One small change is to test bf + complex number errors (you can construct complex Python numbers using `complex(<real>, <imag>)`) and to flip the tests so you're adding the number / other tensor + the bfloat16 tensor. You can do this like this:

```
bf = torch.tensor(5.5, dtype=torch.bfloat16, device=device)
scalars = (2.2, 5, complex(1, -1))
for scalar in scalars:
  if isinstance(scalar, complex):
    with self.assertRaises(RuntimeError):
      a + b
    with self.assertRaises(RuntimeError):
      b + a
  else:
    self.assertEqual(a + b, b + a)
    self.assertEqual((a + b).dtype, 
```

And then in the next section you want to create the tensor of the dtype upfront:

```
for dtype in torch.testing.get_all_dtypes():
  t = torch.tensor(1, device=device, dtype=dtype)
  if dtype in (torch.float32, torch.float64):
    self.assertEqual(bf + t, t + bf)
    self.assertEqual(bf.dtype, dtype)
...
```

How does that sound?

Reply:
Yeah I agree. I adapted your proposal a bit